eclipse-collections/src/main/java/org/eclipse/collections/impl/stack/immutable/ImmutableEmptyStack.java

```diff
@@ -103,7 +103,7 @@ private Object readResolve()
     @Override
     public ImmutableStack<T> push(T item)
     {
-        return new ImmutableNotEmptyStack<T>(item, this);
+        return new ImmutableNotEmptyStack<>(item, this);
```

Comment:
This was a bug in previous versions which I had reported. Glad they fixed it 😄

Reply:
Unfortunately, this inspection is still buggy in some edge cases.